src/java/com/reucon/openfire/plugin/archive/impl/MucMamPersistenceManager.java

```diff
@@ -290,7 +290,7 @@ static Long getMessageIdForStableId( final MUCRoom room, final String value )
             connection = DbConnectionManager.getConnection();
             pstmt = connection.prepareStatement( "SELECT messageId, stanza FROM ofMucConversationLog WHERE messageId IS NOT NULL AND roomID=? AND stanza LIKE ? AND stanza LIKE ?" );
             pstmt.setLong( 1, room.getID() );
-            pstmt.setString( 2, "%"+value+"%" );
+            pstmt.setString( 2, "%id=\""+value+"\"%" );
```

Comment:
This will fail consistently when `'` is used instead of `"`:

```xml
<bar id="foo" />
```
versus

```xml
<bar id='foo' />
```

Reply:
Think I've fixed it, but it's untested.

Reply:
We're well into territory of database optimization here, which I'm not very experienced with, but: my concern is that for most rows (that are not going to match), this would duplicate the text-based search effort. Is that something that we should be concerned of?

Reply:
After chatting to @guusdk, changed this to be more defensive in false positives rather than more precise (+expensive) with the database.